src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -733,23 +678,40 @@ mongoc_topology_destroy (mongoc_topology_t *topology)
       BSON_ASSERT (topology->scanner_state == MONGOC_TOPOLOGY_SCANNER_OFF);
       mongoc_set_destroy (topology->server_monitors);
       mongoc_set_destroy (topology->rtt_monitors);
-      bson_mutex_destroy (&topology->apm_mutex);
       bson_mutex_destroy (&topology->srv_polling_mtx);
       mongoc_cond_destroy (&topology->srv_polling_cond);
    }
 
-   if (topology->valid) {
-      /* Do not emit a topology_closed event. A topology opening event was not
-       * emitted. */
-      _mongoc_topology_description_monitor_closed (mc_tpld_unsafe_get_const (topology));
+   /* Before reporting this topology as closed, life cycle rules expect us to close
```

Comment:
> Added missing end-of-life SDAM events.

Suggest noting this change in the NEWS since this impacts how callbacks are called.

I set `PHP Team Attention Needed` in CDRIVER-4500 to generate a ticket to inform the PHP team. (This required transitioning the ticket to `Ready for Work`, then back to `Code Review` to work around [PROG-12016](https://jira.mongodb.org/browse/PROG-12016).